### PR TITLE
Match several functions in daTbox_c

### PR DESCRIPTION
**Pull request by Sage-of-Mirrors**

### CODE REVIEW

File: src/d/actor/d_a_tbox.cpp [hunk 1]
```diff
@@ -451,23 +530,152 @@ s32 daTbox_c::getFuncType() {
...
-    /* Nonmatching */
+s32 CheckCreateHeap(fopAc_ac_c* i_actor) {
+    return static_cast<daTbox_c*>(i_actor)->CreateHeap();
 }
 
 /* 000010AC-0000114C       .text CreateHeap__8daTbox_cFv */
 s32 daTbox_c::CreateHeap() {
-    /* Nonmatching */
+    if (commonShapeSet() != cPhs_COMPLEATE_e) {
+        return 0;
```
Comment (magcius) [inline src/d/actor/d_a_tbox.cpp:560]: return FALSE

File: src/d/actor/d_a_tbox.cpp [hunk 2]
```diff
@@ -451,23 +530,152 @@ s32 daTbox_c::getFuncType() {
...
+
+    if (checkEnv() && envShapeSet() != cPhs_COMPLEATE_e) {
+        return 0;
+    }
+
+    if (!checkOpen() && effectShapeSet() != cPhs_COMPLEATE_e) {
+        return 0;
+    }
+
+    return bgCheckSet() != 4 ? FALSE : TRUE;
```
Comment (magcius) [inline src/d/actor/d_a_tbox.cpp:571]: != cPhs_COMPLEATE_e

File: src/d/actor/d_a_tbox.cpp [hunk 3]
```diff
@@ -451,23 +530,152 @@ s32 daTbox_c::getFuncType() {
...
+    if (mRoomNo == -1 || mRoomNo == 0x3F) {
+        return FALSE;
+    }
+
+    s32 swNo = getSwNo();
+    if (swNo >= 0xC0) {
+        return FALSE;
+    }
+
+    return dComIfGs_isSwitch(swNo, mRoomNo) ? TRUE : FALSE;
```
Comment (magcius) [inline src/d/actor/d_a_tbox.cpp:549]: for these tails, can you try out e.g.

```cpp
if (dComIfGs_isSwitch(sw, mRoomNo))
    return TRUE;

return FALSE;
```

or the inverse of that? curious to see if it works

File: src/d/actor/d_a_tbox.cpp [hunk 4]
```diff
@@ -390,28 +396,60 @@ s32 daTbox_c::bgCheckSet() {
...
 }
 
 /* 00000BB0-00000C14       .text searchRoomNo__8daTbox_cFv */
 void daTbox_c::searchRoomNo() {
-    /* Nonmatching */
+    if (mRoomNo == -1) {
+        mRoomNo = orig.angle.x & 0x3F;
+    }
+
+    if ((m03F0 & 0x02) != 0) {
```
Comment (LagoLunatic) [inline src/d/actor/d_a_tbox.cpp:409]: This is probably done with `flagCheck__8daTbox_cFUs` from the debug map

File: src/d/actor/d_a_tbox.cpp [hunk 5]
```diff
@@ -451,23 +530,152 @@ s32 daTbox_c::getFuncType() {
...
+                    mActionFunc = actionSwOnWait2;
+                }
+
+                m034C = 2.0f;
+
+                J3DFrameCtrl* frameCtrl = mpAppearRegAnm->getFrameCtrl();
+                frameCtrl->setFrame(frameCtrl->getEnd());
+            }
+            else {
+                 m03F0 |= 0x04;
```
Comment (LagoLunatic) [inline src/d/actor/d_a_tbox.cpp:613]: `flagOn__8daTbox_cFUs`

File: src/d/actor/d_a_tbox.cpp [hunk 6]
```diff
@@ -421,22 +459,63 @@ modelInfo& daTbox_c::getModelInfo() {
...
 
 /* 00000ECC-00000F8C       .text surfaceProc__8daTbox_cFv */
 void daTbox_c::surfaceProc() {
-    /* Nonmatching */
+    if (mpBgWCurrent != NULL && (m03F0 & 0x20) != 0) {
+        if (m03EC < -1.0f) {
+            m03EC += 1.0f;
+        }
+        else {
+            m03F0 &= -33;
```
Comment (LagoLunatic) [inline src/d/actor/d_a_tbox.cpp:500]:
```suggestion
            flagOff(0x20);
```

**Pull request merged**